src/main/src/main/java/org/geoserver/catalog/util/ReaderDimensionsAccessor.java

```diff
@@ -295,14 +366,19 @@ public boolean hasElevation() throws IOException {
         if (!hasElevation()) {
             return null;
         }
-        // parse the values from the reader, they are exposed as strings...
-        String[] elevationValues = reader.getMetadataValue(ELEVATION_DOMAIN).split(",");
-        TreeSet<Object> elevations = new TreeSet<>(ELEVATION_COMPARATOR);
-        for (String val : elevationValues) {
-            try {
-                elevations.add(parseNumberOrRange(val));
-            } catch (Exception e) {
-                LOGGER.log(Level.WARNING, e.getMessage(), e);
+        TreeSet<Object> elevations = null;
+        if (dimensionInfo != null && dimensionInfo.getFixedValues() != null) {
+            elevations = getFixedValueRange(dimensionInfo.getFixedValues());
```

Comment:
Again, you are simply populating the a TreeSet with the string values in the fixed values string. I don't thik this is what we need. If you see down in the else statement the values are actually parsed before being added.